### ISSUE TITLE
DataDaylighting::MaxRefPoints is obsolete

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#8846]
Issue overview
--------------
Daylighting:Controls was extended a while back to allow more than two reference points, but `DataDaylighting::  constexpr int MaxRefPoints(2)` is still in use in several places.

Output:IlluminanceMap lists the reference points in the zone as part of the map title - this should be using` dataDaylightingData->ZoneDaylight(ZoneNum).TotalDaylRefPoints`
(This might change to `SpaceDaylight...`)

In `CalcDayltgCoeffsMapPoints`, when zeroing some of the arrays [here](https://github.com/NREL/EnergyPlus/blob/1a131475d56bc6ff761569b794799287a3e0b581/src/EnergyPlus/DaylightingManager.cc#L1359-L1406), should be using `state.dataDaylightingData->IllumMapCalc(MapNum).TotalMapRefPoints` instead of `MaxRefPoints`.

At least that's what it seems from looking at the code.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.6.0-1a131475d56bc6ff761569b794799287a3e0b581
 - Noticed while working on #8394 

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file use example file DaylightingDeviceTubular and add some extra daylighting reference points and run with detailed timestep shadowing calculations.
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by RKStrand** [comment]
@mjwitte Interesting.  In the ...RefPoints routine above ...MapPoints, this seems like it was done correctly.  So, if I am understanding this, similar changes need to be made to ...MapPoints to bring it inline with ...RefPoints.  Is it fair to say that MaxRefPoints should be eliminated from the code then?

**Comment by mjwitte** [comment]
Not 100% certain, but `MaxRefPoints` can either be eliminated, or it should be set based on the maximum number of reference points in any given zone? Or is that already in a different variable?

**Comment by RKStrand** [comment]
I wasn't 100% clear in my last comment.  What I should have said was: MaxRefPoints should be replaced with the new variable (state.dataDaylightingData->ZoneDaylight(ZoneNum).TotalDaylRefPoints) that allows more than 2 points and then eventually once all of those happen MaxRefPoints can be removed from the code since it isn't used anymore.

**Comment by amirroth** [comment]
Thanks. We are beginning a pass to convert some of the ObjexxFCL Arrays whose sizes are known at compile time (i.e., `constexpr int`) to `std::array`. Arrays with `MaxRefPoint` dimensions were on the docket, but I will remove them now.

**Comment by RKStrand** [comment]
Correction...I think the variable in this routine is actually:

state.dataDaylightingData->IllumMapCalc(MapNum).TotalMapRefPoints

But I'll double check that as I work on this...

**Comment by RKStrand** [comment]
Wow...adding an additional reference point actually causes a hard crash. 😳  Amazing that no one has run into this yet.

**Comment by mjwitte** [comment]
You mean >2 reference points in a zone with an illuminance map? Or >2 reference points, even without the map?

**Comment by RKStrand** [comment]
With an Illuminance map--I haven't tried to take out the illuminance map to see if it still happens.  In Daylighting:Controls, I added a third reference point.  As soon as I did that, I got a hard crash at line 9988 in ReportIllumMap of DaylightingManager.cc.  I was expecting it, because it is still using MaxRefPoints to allocate an array. However, the total reference points now exceeds 2 because I added the third point.  When it gets to the array index of 3 that was only allocated to 2, E+ crashes.
